javascript/javascript1/week2/main.js

```diff
@@ -0,0 +1,184 @@
+            console.log("Please, enter a temperature");
+        }
+        return temp;
+    }
+    const clothesToWear = getDressed();
+
+//Student manager
+const class07Students = [];
+
+
+function addStudentToClass(studentName) {
+
+    if (!class07Students.includes(studentName) // excludes duplicates
+    && studentName != ``) { //  and does not add an empty string
+        class07Students.push(studentName);
+    } else if (class07Students.includes(studentName)) { // if includes duplicats does not add it
+        console.log(`Student ${studentName} is already in the class`);
+    }
+
+    if (class07Students.length <= 6 
+        && studentName === "Queen") { // always space for Queen
+            class07Students.push();
+        } else if (class07Students.length > 6 
+        && studentName !== "Queen") { // otherwise max.length = 6
+            class07Students.length = 6;
+            console.log("Cannot add more students to class 07");
+        } 
+
+    return studentName;
+}
+
+addStudentToClass(""); 
+addStudentToClass("Anna"); // 1
+addStudentToClass("Maria"); // 2
+addStudentToClass("Viktor"); // 3
+addStudentToClass("Peter"); // 4
+addStudentToClass("Viktoria"); // 5
+addStudentToClass("Nelia"); // 6
+addStudentToClass("Christian"); // Cannot add more students to class 07
+addStudentToClass("Mathilda"); // Cannot add more students to class 07
+addStudentToClass("Peter"); //  Student Peter is already in the class
+addStudentToClass("Queen"); // adds Queen
+
+
+console.log(class07Students)
+
+function getNumberOfStudents(number) {
+  number = class07Students.length;
+  return number;
+}
+console.log(getNumberOfStudents());
+
+// Candy helper 
+const boughtCandyPrices = []; //array to store prices of different candies
+
+//price list for 1g
+const sweet = 0.5;
+const chocolate = 0.7;
+const toffee = 1.1;
+const chewingGum = 0.03;
+
+function addCandy(candyType = sweet, weight = 10) { //default value of parameters
+    result = candyType * weight; // price for selected weight;
+    return boughtCandyPrices.push(result); // push result to array
+}
+//add some sweets to array
+addCandy(toffee, 20);
+addCandy(chocolate, 30);
+addCandy(chewingGum, 10);
+console.log(boughtCandyPrices); // prices in array: 22, 21, 0.3
+
+let amountToSpend = Math.random() * 100; // random number assigned to value we can spend
+console.log(`You can spend up to $${amountToSpend.toFixed(2)}`); // round value to 2 decimals
+
+let sum = 0;
+let i = 0;
+while (i < boughtCandyPrices.length) { // while loop to loop through an array
+    sum = sum + boughtCandyPrices[i]; // each array element would be summed up to the value of sum
+    i++;
+}
+
+console.log(`It will cost $${sum}`); // sum is 43.3
+
+function canBuyMoreCandy(boughtCandyPrices) { // check if sum <, <= amountToSpend
+
+    sum < amountToSpend ? 
+    console.log("You can buy more, so please do") : 
+    console.log("Enough candy for you!");
+
+    return boughtCandyPrices;
+}
```

Comment:
It is really great that you used `while` in here. Still, I think you should put this loop inside of a function, so that you can actually check if you can buy more candies on every function call